File: spec/php/StrLiteralsTest.php
<?php
namespace Kaitai\Struct\Tests;

class StrLiteralsTest extends TestCase {
    public function testStrLiterals() {
        $r = StrLiterals::fromFile(self::SRC_DIR_PATH . "/fixed_struct.bin");

        $this->assertSame("\u{0}\u{1}\u{2}\u{7}\u{8}\u{a}\u{d}\u{9}\u{b}\u{c}\u{1b}\u{3d}\u{7}\u{a}\u{24}\u{263b}", $r->complexStr);
        $this->assertSame("\u{22}\u{22}\u{22}", $r->doubleQuotes);
        $this->assertSame("\u{5c}\u{5c}\u{5c}", $r->backslashes);
        $this->assertSame("\u{0}\u{32}\u{32}", $r->octalEatup);
        $this->assertSame("\u{2}\u{32}", $r->octalEatup2);
    }
}

File: spec/go/str_literals_test.go
package spec

import (
	"runtime/debug"
	"os"
	"testing"
	"github.com/kaitai-io/kaitai_struct_go_runtime/kaitai"
	. "test_formats"
	"github.com/stretchr/testify/assert"
)

func TestStrLiterals(t *testing.T) {
	defer func() {
		if r := recover(); r != nil {
			debug.PrintStack()
			t.Fatal("unexpected panic:", r)
		}
	}()
	f, err := os.Open("../../src/fixed_struct.bin")
	if err != nil {
		t.Fatal(err)
	}
	s := kaitai.NewStream(f)
	var r StrLiterals
	err = r.Read(s, &r, &r)
	if err != nil {
		t.Fatal(err)
	}

	tmp1, err := r.ComplexStr()
	if err != nil {
		t.Fatal(err)
	}
	assert.EqualValues(t, []rune{0, 1, 2, 7, 8, 10, 13, 9, 11, 12, 27, 61, 7, 10, 36, 9787}, []rune(tmp1))
	tmp2, err := r.DoubleQuotes()
	if err != nil {
		t.Fatal(err)
	}
	assert.EqualValues(t, []rune{34, 34, 34}, []rune(tmp2))
	tmp3, err := r.Backslashes()
	if err != nil {
		t.Fatal(err)
	}
	assert.EqualValues(t, []rune{92, 92, 92}, []rune(tmp3))
	tmp4, err := r.OctalEatup()
	if err != nil {
		t.Fatal(err)
	}
	assert.EqualValues(t, []rune{0, 50, 50}, []rune(tmp4))
	tmp5, err := r.OctalEatup2()
	if err != nil {
		t.Fatal(err)
	}
	assert.EqualValues(t, []rune{2, 50}, []rune(tmp5))
}
